builtin/providers/vcd/resource_vcd_firewall_rules.go

```diff
@@ -0,0 +1,241 @@
+package vcd
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hmrc/vmware-govcd"
+	types "github.com/hmrc/vmware-govcd/types/v56"
+	"strings"
+)
+
+func resourceVcdFirewallRules() *schema.Resource {
```

Comment:
Ah sure that makes sense. So, if rule order is significant you probably want to change to `schema.TypeList` for your list of rules. `schema.TypeSet` does not define any specific order for its contents.